lib/processor.js

```diff
@@ -9,179 +9,51 @@
+            traverse(node.children[i], callbacks);
         }
-
-        block = result[result.length - 1];
-        regexp = [
-            "^ {0,3}",
-            block.fence[0],
-            "{",
-            block.fence.length,
-            ",}\\h*(?:\\r?\\n|$)"
-        ].join("");
-        return match(new RegExp(regexp));
     }
-
-    /**
-     * Starts a new block and pushes it onto the result list.
-     * @param {string} indent The indent used for this block.
-     * @param {string} fence The backticks or tildes used to start the block.
-     * @returns {void}
-     */
-    function startBlock(indent, fence) {
-        result.push({
-            range: [index],
-            loc: {
-                start: {
-                    line: line,
-                    column: indent.length
-                }
-            },
-            indent: indent,
-            fence: fence
-        });
-
-        mode = modes.CODE;
-    }
-
-    /**
-     * Finishes the most recent block.
-     * @param {number} backtrack How far behind `index` the block ends.
-     * @returns {void}
-     */
-    function finishBlock(backtrack) {
-        var block = result[result.length - 1];
-
-        block.range[1] = index - backtrack;
-        block.loc.end = {
-            line: line,
-            column: column || block.indent.length
-        };
-        block.rawText = text.slice(block.range[0], block.range[1]);
-
-        if (block.indent.length > 0) {
-            block.text = block.rawText.replace(new RegExp("^ {1," + block.indent.length + "}", "gm"), "");
-        } else {
-            block.text = block.rawText;
-        }
-
-        mode = modes.NORMAL;
-    }
-
-    while (index < text.length) {
-
-        // Newline
-        if (match(/^\r?\n/)) {
-            line += 1;
-            column = 0;
-
-        // Opening code fence
-        } else if ((next = matchOpeningCodeFence())) {
-            line += 1;
-            column = 0;
-
-            startBlock(next.indent, next.fence);
-
-        // Closing code fence
-        } else if ((next = matchClosingCodeFence())) {
-            finishBlock(next[0].length);
-
-            line += 1;
-            column = 0;
-
-        // Line content
-        } else {
-            next = match(/^[^\r\n]*/);
-            column += next[0].length;
-        }
-    }
-
-    // Finish an unclosed code fence
-    if (mode === modes.CODE) {
-        finishBlock(0);
-    }
-
-    return result;
 }
 
 /**
- * Extracts code fences from Markdown text.
+ * Extracts lintable JavaScript code blocks from Markdown text.
  * @param {string} text The text of the file.
  * @returns {[string]} Code blocks to lint.
  */
 function preprocess(text) {
-    blocks = parse(text);
+    var ast = remark.parse(text);
+
+    blocks = [];
+    traverse(ast, {
+        "code": function(node) {
+            if (node.lang && SUPPORTED_SYNTAXES.indexOf(node.lang.toLowerCase()) >= 0) {
+                blocks.push(node);
```

Comment:
Why not `blocks.push(node.value)` directly if you're not using any other property of the node afterwards?